nestkernel/kernel_manager.cpp

```diff
@@ -132,25 +130,18 @@ nest::KernelManager::reset()
 void
 nest::KernelManager::change_number_of_threads( thread new_num_threads )
 {
-  node_manager.finalize();
-  connection_manager.finalize();
-  model_manager.finalize();
-  modelrange_manager.finalize();
-  random_manager.finalize();
+  // For descriptions of the following conditions, see the cascade of
+  // user input checks in VPManager::set_status()
+  assert( node_manager.size() == 0 );
```

Comment:
Is the meaning of the comment above that any user input is checked and problems properly handled in `VPManager::set_status()`, so we only need the assertions here as a extra safety check? And if a user tries to change the number of threads at a time it cannot be changed they will get a proper error message, not NEST crashing on an assertion?

Reply:
Yes. There are user-level check in `VPManager::set_status()` and the assertions here are only there to make sure those are properly checking their thing.

Reply:
Thanks, I will create a suggestion for a clearer comment.